packages/babel-core/src/config/config-chain.js

```diff
@@ -209,16 +229,30 @@ export function* buildRootChain(
     }
 
     if (babelrcFile) {
+      const validatedFile = validateBabelrcFile(babelrcFile);
+      const babelrcLogger = new ConfigPrinter();
       const result = yield* loadFileChain(
-        validateBabelrcFile(babelrcFile),
+        validatedFile,
         context,
+        undefined,
+        babelrcLogger,
       );
       if (!result) return null;
+      babelRcReport = babelrcLogger.output();
 
       mergeChain(fileChain, result);
     }
   }
 
+  if (context.showConfig) {
+    console.log(
+      // print config by the order of descending priority
```

Comment:
Maybe we should print in the output that in case of conflicts the first config overrides the second?

Reply:
I tend to keep the output succinct. The implementation detail of config merging should be referenced from babel docs but not a debug output.

To reason how these config merges, we can print the resolved config in later PRs. Think of the `Styles` tab v.s. the `Computed` tab in the Chrome DevTools / Elements.